src/openapi/authProvider.js

```diff
@@ -0,0 +1,302 @@
+/**

```

Comment:
This was pulled out of transportAuth

Its basically the Auth class from inside there, but with token management and transport related concerns left in transportAuth

Reply:
(apart from this there are no functional changes in this file)